frontendVue3/src/components/FactLabel.vue

```diff
@@ -0,0 +1,1120 @@
+            "value": [],
+            "size": 0,
+            "ratio": 0.0
+        },
+        {
+            "name": "Configurations",
+            "description": "Number of configurations represented by the feature model. If <= is shown, the number represents an upper estimation bound.",
+            "parent": null,
+            "level": 0,
+            "value": "26256",
+            "size": null,
+            "ratio": null
+        }
+    ]
+}
+const FM_CHAR_NAME_DESC = "Name"; // FM Characterization Name Descriptor
+const FM_CHAR_HREF_DESC= "Reference"
+const FM_CHAR_DESC_DESC="Description"
+export default {
+    name: 'FactLabel',
+
+    components: {},
+
+    props: {},
+
+    data: () => ({
+        name: "",
+        fmHref:"",
+        desc:"",
+        expandedRoot: [],
+        expandedSubs: [],
+        expandedSubSubs: [],
+        factHeaders: [{ key: "name", sortable: false }, { key: "value", sortable: false }],
+        expandableHeaders: [{ key: 'data-table-expand' }, { key: "name", sortable: false }, { key: "value", sortable: false }],
+        metadata: [], // Array of simple k,v pairs
+        metrics: [],
+        analysis: [],
+    }),
+
+
+    computed: {},
+    created() {
+        this.initialize()
+    },
+    methods: {
+        initialize() {
+            this.fillMetaData();
+            this.fillAnalysis();
+            this.fillMetrics();
+            return;
+        },
+        fillMetaData() {
+            this.metadata = facts.metadata.filter((entry) => {
+                if (entry.name === FM_CHAR_NAME_DESC) {
+                    this.name = entry.value; //handle special entry Name which defines Name of FM
+                    return false;
+                }else if (entry.name === FM_CHAR_HREF_DESC) {
+                    this.fmHref = entry.value; //handle special entry Name which defines Name of FM
+                    return false;
+                }else if (entry.name === FM_CHAR_DESC_DESC) {
+                    this.desc = entry.value; //handle special entry Name which defines Name of FM
+                    return false;
+                }else{
+                    return true;
+                }
+            });
+            this.metadata = this.metadata.map((entry) => {
+                var obj = {}; // temporry JSON object to fill for visualisation of metadata
+                obj["name"] = entry.name;
+                obj["value"] = this.getDisplayValue(entry);
+                return obj;
+            });
+        },
+
+        fillAnalysis() {
+            this.analysis = facts.analysis.map((entry) => {
+                var obj = {}; // temporry JSON object to fill for visualisation of analysis
+                obj["name"] = entry.name;
+                // Value depends on if Size and Ratio are set
+
+                obj["value"] = this.getDisplayValue(entry);
+                return obj;
+            });
+        },
+        fillMetrics() {
+            if (this.getMaxLevel > 2) {
```

Comment:
Currently not happy with the solution but did not want to overengineer it since i dont thinks a nesting level higher than 3 is suitable for a fact labe :laughing: